opsmngr/logs.go

```diff
@@ -0,0 +1,285 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opsmngr
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	atlas "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+)
+
+const (
+	logsBasePath = "groups/%s/logCollectionJobs"
+)
+
+// LogCollectionService is an interface for interfacing with the Log Collection Jobs
+// endpoints of the MongoDB Ops Manager API.
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/log-collection/
+type LogCollectionService interface {
+	List(context.Context, string, *LogListOptions) (*LogCollectionJobs, *atlas.Response, error)
+	Get(context.Context, string, string, *LogListOptions) (*LogCollectionJob, *atlas.Response, error)
+	Retry(context.Context, string, string) (*atlas.Response, error)
+	Create(context.Context, string, *LogCollectionJob) (*LogCollectionJob, *atlas.Response, error)
+	Extend(context.Context, string, string, *LogCollectionJob) (*atlas.Response, error)
+	Delete(context.Context, string, string) (*atlas.Response, error)
+}
+
+// LogsService is an interface for interfacing with the Log Collection Jobs
+// endpoints of the MongoDB Ops Manager API.
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/log-collection/
+type LogsService interface {
+	Download(context.Context, string, string, io.Writer) (*atlas.Response, error)
+}
+
+// LogCollectionServiceOp handles communication with the Log Collection Jobs related methods of the
+// MongoDB Ops Manager API
+type LogCollectionServiceOp struct {
+	Client atlas.RequestDoer
+}
+
+// LogCollectionServiceOp handles communication with the Log Collection Jobs download method of the
+// MongoDB Ops Manager API
+type LogsServiceOp struct {
+	Client atlas.GZipRequestDoer
+}
+
+var _ LogCollectionService = &LogCollectionServiceOp{}
+var _ LogsService = &LogsServiceOp{}
+
+// LogCollectionJob represents a Log Collection Job in the MongoDB Ops Manager API.
+type LogCollectionJob struct {
+	ID                         string       `json:"id,omitempty"`
+	GroupID                    string       `json:"groupId,omitempty"`
+	UserID                     string       `json:"userId,omitempty"`
+	CreationDate               string       `json:"creationDate,omitempty"`
+	ExpirationDate             string       `json:"expirationDate,omitempty"`
+	Status                     string       `json:"status,omitempty"`
+	ResourceType               string       `json:"resourceType,omitempty"`
+	ResourceName               string       `json:"resourceName,omitempty"`
+	RootResourceName           string       `json:"rootResourceName,omitempty"`
+	RootResourceType           string       `json:"rootResourceType,omitempty"`
+	URL                        string       `json:"downloadUrl,omitempty"`
+	Redacted                   *bool        `json:"redacted"`
```

Comment:
My previous comment was that one or the other, 

either use a pointer and `omitempty`, something like
```go
Redacted                   *bool        `json:"redacted,omitempty"`
```

 or don't use a pointer and remove `omitempty`
```go
Redacted                   bool        `json:"redacted"`
```